satpy/readers/clavrx.py

```diff
@@ -94,9 +108,29 @@ def _get_rows_per_scan(sensor: str) -> Optional[int]:
     return None
 
 
+def _scale_data(data_arr: Union[xr.DataArray, int], scale_factor: float, add_offset: float) -> xr.DataArray:
+    """Scale data, if needed."""
+    scaling_needed = not (scale_factor == 1.0 and add_offset == 0.0)
+    if scaling_needed:
+        data_arr = data_arr * scale_factor + add_offset
+    return data_arr
+
+
 class _CLAVRxHelper:
     """A base class for the CLAVRx File Handlers."""
 
+    @staticmethod
+    def _get_nadir_resolution(sensor, resolution_from_filename_info):
+        """Get nadir resolution."""
+        for k, v in NADIR_RESOLUTION.items():
+            if sensor.startswith(k):
+                return v
+        res = resolution_from_filename_info
+        if res.endswith("m"):
+            return int(res[:-1])
+        elif res is not None:
```

Comment:
`elif` is unnecessary. Not sure why the linter didn't complain about this. Maybe it isn't enabled on satpy or ruff isn't checking it yet.

```suggestion
        if res is not None:
```

But also...if `res` can be `None` then won't the previous `res.endswith` fail?

Reply:
Yes.  It would.

Reply:
Other than setting global attributes, is the nadir resolution used in remapping?  I am wondering if it is okay if the nadir resolution is returned as none?  There should be enough here to always have a result.  I am just wondering if I should just let the reader fail if there is a None value.

Reply:
It looks like `self.resolution` is used to fill in `ds_info["resolution"]` in the datasets. This becomes part of the `DataArray.attrs` but also part of the `DataID` that Satpy uses to identify each dataset/product. I think it is allowed to be None in most cases when it is unknown or not applicable.

Reply:
This is an odd mix.  Not sure if I should also re-work how the resolution value is extracted from the string.  Sometimes I get too fancy, but I was thinking, hypothetically, resolution could be a different unit system.  It is highly unlikely, but not impossible.

Is this overboard?
```python
def _get_nadir_resolution(sensor, resolution_from_filename_info):
        """Get nadir resolution."""
        for k, v in NADIR_RESOLUTION.items():
            if sensor.startswith(k):
                return v
        if resolution_from_filename_info is None:
            return None
        elif isinstance(resolution_from_filename_info, int):
            return int(resolution_from_filename_info)
        elif isinstance(resolution_from_filename_info, str):
            print(resolution_from_filename_info)
            result = ''.join([i for i in resolution_from_filename_info if i.isdigit()])
            return int(result)
```

Reply:
A little simpler:

```python
    def _get_nadir_resolution(sensor, resolution_from_filename_info):
        """Get nadir resolution."""
        for k, v in NADIR_RESOLUTION.items():
            if sensor.startswith(k):
                return v
        if resolution_from_filename_info is None:
            return None
        if isinstance(resolution_from_filename_info, str):
            resolution_from_filename_info = ''.join([i for i in resolution_from_filename_info if i.isdigit()])
        return int(resolution_from_filename_info)
```

Right now Satpy doesn't support units other than meters in the resolution field...although it also doesn't use it beyond the user requesting a specific resolution. I'd say removing the "m" is probably good enough unless you have evidence that a "123 degrees" or something might exist. Oh I guess there could be "km" but in that case you need to actually convert that or specify the units when assigning the resolution in some way. So maybe it is best to let it fail automatically by only removing the "m" (`if isinstance(resolution_from_filename_info, str) and resolution_from_filename_info.startswith("m"):`)